### DISCUSSION TITLE
Build a benchmark for sotopia!

### DISCUSSION THREAD

**Discussion by XuhuiZhou** [sotopia-lab/sotopia#36] [Ideas]
We are currently working on building a benchmark for Sotopia!

@clementou is currently in charge of adapting the Sotopia codebase to accommodate as many models as possible!
We are looking for help in designing the UI for the Sotopia benchmark.

Some inspirations for the UI:
* https://huggingface.co/spaces/allenai/WildBench
* https://huggingface.co/spaces/lmsys/mt-bench

**Comment by Algoroxyolo** [comment]
what models are we looking at?

I am thinking about a few: qwen series, Mixtral8x7b, llama2 series

**Reply by XuhuiZhou** [reply]
@clementou

**Reply by Algoroxyolo** [reply]
Also- do you mind if I get a few more people of interest involved?

**Reply by Algoroxyolo** [reply]
Can I know more about current progress?

**Reply by clementou** [reply]
Switched to using the [LiteLLM Library](https://docs.litellm.ai/docs/providers) so we can use any API endpoints supported there.

Having some issues right now [calculating model performance](https://github.com/sotopia-lab/sotopia/commit/96024a1784cf5126361ed7ceca81d57dc6d86d55) since writing episodes still uses the model_dict to write the model name. So the model name written to the episode doesn't match the model being used. Can't pinpoint where this happens right now, but a test case for this would help.

Haven't started on the UI yet but should be fairly easy.

**Reply by XuhuiZhou** [reply]
Thanks @clementou! Maybe @Algoroxyolo you could start working on the UI?

**Reply by Algoroxyolo** [reply]
I am not so experienced with UI but I can try

**Reply by Algoroxyolo** [reply]
Where should I start looking at?

**Reply by XuhuiZhou** [reply]
Here https://huggingface.co/spaces/allenai/WildBench!

**Comment by XuhuiZhou** [comment]
> Also- do you mind if I get a few more people of interest involved?

@Algoroxyolo 
Yeah! As long as they are happy to contribute :)
